### ISSUE TITLE
high-DPI detection of AWTTerminalFontConfiguration is broken on my system

### ISSUE THREAD

**Issue by jrb0001** [mabe02/lanterna#351]
I have two monitors at 1920x1080 each, but Java reports them as `[3840, 1080]` and `[1920, 1080]` which triggers the extremely high font size. Wouldn't it be a better idea to use `Toolkit.getDefaultToolkit().getScreenResolution()` and then maybe 160 as threshold? If that's not possible, maybe the combination of `width > 2000 && height > 1400`?

**Comment by avl42** [comment]
I'm not sure I understand it correctly, but Lanterna might have a hard time working around Java problems.  If you have a particular workaround in mind, where Lanterna could indeed solve it, then a PR would be welcome, provided it doesn't introduce problems for "normal" screen setups.

**Comment by jrb0001** [comment]
Changing something without knowing why it was done in the current way is a bit meh. Is there any particular reason why it was done the current way or was it simply the first idea?

**Comment by avl42** [comment]
I can only guess: maybe historical experience with queried screen
resolutions were bad enough to rather rely on other heuristics.

Maybe nowadays DPI values are better quality nowadays and beat other
heuristics.  They're probably better on mobile devices (phones, tablets)
where the driver "knows" the physical screen, and still mostly guesswork
where the driver doesn't know the physical screen.

So much for my speculation about status-quo ;-)

**Comment by avl42** [comment]
One of my machines' graphics port is connected to a VGA-splitter that
drives both a 22" monitor and a 40" TV-screen.  What would be a correct DPI
value?

**Comment by mabe02** [comment]
Relevant code (from `AWTTerminalFontConfiguration`):
```
        // Assume the first GraphicsDevice is the primary screen (this isn't always correct but what to do?)
        // Warning, there could be printers coming back here according to JavaDoc! Hopefully Java is reasonable and
        // passes them in after the real monitor(s).
        if (gs.length > 0) {
            int primaryMonitorWidth = gs[0].getDisplayMode().getWidth();

            // If the width is wider than Full HD (1080p, or 1920x1080), then assume it's high-DPI
            if (primaryMonitorWidth > 2000) {
                CHOSEN_FONT_SIZE = 28;
            }
        }

        // If no size was picked, default to 14
        if(CHOSEN_FONT_SIZE == null) {
            CHOSEN_FONT_SIZE = 14;
        }
```
This is pretty simple, yes. I know on Windows, Oracle has made the java.exe binary report itself as HiDPI-aware which means the OS won't apply any scaling when on HiDPI monitors. Not sure about Mac but I suspect it works similarly. Linux-based systems (like Ubuntu) probably don't do any scaling for generic X applications. So for these reasons, I chose the approach above. Doesn't work great with multi-monitor setups, agreed. The threshold for a width of 2000 was because I figured from 2K monitors and above, you'd want the larger font.

Now I'm pretty sure for your own application, you can override this font size to whatever you want. The code above is just for the default value.

**Comment by avl42** [comment]
Isn't the point here that @jrb0001's monitors are **not** "HiDPI" at all,  but instead, Java reports the combined rectangle of monitors as first element of that array, mis-triggering lanterna into believing
in some HiDPI-monitor...

**Comment by mabe02** [comment]
Okay, I didn't think of it that way. You're seeing the combined resolution across the monitors is what Java reports as the primary display's resolution? Yeah, that would work very well...
I actually only have a single-monitor setup at home so I don't have anything readily setup to test this further.

**Comment by jrb0001** [comment]
I should have worked on that a long time ago already but always found an excuse. I am going to try to fix this now.

**Comment by jrb0001** [comment]
Closed by #369.
